### PR TITLE
Refactor Service Brokers configuration menu

### LINKED ISSUE #716: Merge ServiceCatalog and ServiceBrokerList
The [ServiceCatalog component](https://github.com/kubeapps/kubeapps/blob/69b1c0d34fee0770dd055fd284022d8e8ac74f6d/dashboard/src/components/Config/ServiceCatalog/index.tsx) is not doing much and can probably be merged into the ServiceBrokerList component.

Putting this into the backlog as an extra task for the Service Catalog refactor.

### PR DESCRIPTION
**Pull request by andresmgot**
Ref #743 
Closes #716

Refactor how Service Brokers are represented in the configuration menu:

 - Use the InfoCard format
 - Use the latest niceties (`ErrorSelector`, `LoadingWrapper`)
 - Split `ServiceBrokerItem` into its own component
 - Remove the ServiceCatalog component 
 - Unit tests

TODO:
  - [x] Write tests for SyncButton.tsx
  ~~- [ ] Write tests for ServiceCatalog.tsx~~

Screenshot:

screen shot 2018-10-18 at 15 16 09

For the record, this was the status before:

screen shot 2018-10-18 at 18 07 21

### CODE REVIEW

**Comment by arapulido** [comment]
Looking great!

**Comment by andresmgot** [comment]
This is now ready to be reviewed @prydonius @migmartri

File: dashboard/src/components/Config/ServiceBrokerList/ServiceBrokerItem.tsx [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  parser.href = brokerURL;
+  if (parser.hostname.match(/azure/)) {
+    return azureIcon;
+  } else if (parser.hostname.match(/servicebroker\.googleapis/)) {
+    return gcpIcon;
+  }
+  return placeholder;
+}
+
+const ServiceBrokerItem: React.SFC<IServiceBrokerItemProps> = props => {
```
Comment (migmartri) [inline dashboard/src/components/Config/ServiceBrokerList/ServiceBrokerItem.tsx]: Thanks for using the stateless component.

File: dashboard/src/components/Config/ServiceBrokerList/ServiceBrokerItem.tsx [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  const { broker, sync } = props;
+  return (
+    <InfoCard
+      title={broker.metadata.name}
+      info={`Last updated ${broker.status.lastCatalogRetrievalTime}`}
```
Comment (migmartri) [inline dashboard/src/components/Config/ServiceBrokerList/ServiceBrokerItem.tsx]: It would be great to humanize this in another patch by using moment.js or equivalent.

File: dashboard/src/containers/ServiceCatalogContainer/ServiceCatalogContainer.ts [hunk 3]
```diff
@@ -3,7 +3,7 @@ import { Action } from "redux";
 import { ThunkDispatch } from "redux-thunk";
```
Comment (prydonius) [inline dashboard/src/containers/ServiceCatalogContainer/ServiceCatalogContainer.ts]: Can we rename this container to ServiceBrokerListContainer?

**Comment by andresmgot** [comment]
> In the screenshot, it looks like the button size has changed. It looks a little big to me, what do you think?

The button is the same size (the card is smaller) but I can make the button smaller.

> can we rename the ServiceCatalogContainer to ServiceBrokerListContainer as it should be?

+1
